src/equations/shallow_water_two_layer_2d.jl

```diff
@@ -0,0 +1,850 @@
+variables. 
+
+Further details are available in the paper:
+- Ulrik Skre Fjordholm (2012)
+Energy conservative and stable schemes for the two-layer shallow water equations.
+(https://doi.org/10.1142/9789814417099_0039)
+"""
+@inline function flux_es(u_ll, u_rr,
+                         orientation_or_normal_direction, 
+                         equations::TwoLayerShallowWaterEquations2D)                   
+  # Compute entropy conservative flux but without the bottom topography
+  f_ec = flux_fjordholm_etal(u_ll, u_rr,
+                            orientation_or_normal_direction,
+                            equations)[1:6]
+
+  # Get maximum signal velocity
+  λ = max_abs_speed_naive(u_ll, u_rr, orientation_or_normal_direction, equations)
+
+  # Get entropy variables but without the bottom topography
+  q_rr = cons2entropy(u_rr,equations)[1:6]
+  q_ll = cons2entropy(u_ll,equations)[1:6]
+
+  # Average values from left and right 
+  u_avg = (u_ll + u_rr)/2
+
+  # Introduce variables for better readability
+  rho1 = equations.rho1
+  rho2 = equations.rho2
+  g    = equations.gravity
+  drho = rho1 - rho2
+
+  # Entropy Jacobian matrix
+  H = [[-rho2/(g*rho1*drho);;
+        -rho2*u_avg[2]/(g*rho1*u_avg[1]*drho);;
+        -rho2*u_avg[3]/(g*rho1*u_avg[1]*drho);;
+        1.0/(g*drho);;
+        u_avg[5]/(g*u_avg[4]*drho);;
+        u_avg[6]/(g*u_avg[4]*drho)];
+      [-rho2*u_avg[2]/(g*rho1*u_avg[1]*drho);;
+        (g*rho1*u_avg[1]^3 - g*rho2*u_avg[1]^3 - rho2*u_avg[2]^2)/(g*rho1*u_avg[1]^2*drho);;
+        -rho2*u_avg[2]*u_avg[3]/(g*rho1*u_avg[1]^2*drho);;
+        u_avg[2]/(g*u_avg[1]*drho);;
+        u_avg[2]*u_avg[5]/(g*u_avg[1]*u_avg[4]*drho);;
+        u_avg[2]*u_avg[6]/(g*u_avg[1]*u_avg[4]*drho)];
+      [-rho2*u_avg[3]/(g*rho1*u_avg[1]*drho);;
+        -rho2*u_avg[2]*u_avg[3]/(g*rho1*u_avg[1]^2*drho);;
+        (g*rho1*u_avg[1]^3 - g*rho2*u_avg[1]^3 - rho2*u_avg[3]^2)/(g*rho1*u_avg[1]^2*drho);;
+        u_avg[3]/(g*u_avg[1]*drho);;
+        u_avg[3]*u_avg[5]/(g*u_avg[1]*u_avg[4]*drho);;
+        u_avg[3]*u_avg[6]/(g*u_avg[1]*u_avg[4]*drho)];
+      [1.0/(g*drho);;
+        u_avg[2]/(g*u_avg[1]*drho);;
+        u_avg[3]/(g*u_avg[1]*drho);;
+      -1.0/(g*drho);;
+      -u_avg[5]/(g*u_avg[4]*drho);;
+      -u_avg[6]/(g*u_avg[4]*drho)];
+      [u_avg[5]/(g*u_avg[4]*drho);;
+        u_avg[2]*u_avg[5]/(g*u_avg[1]*u_avg[4]*drho);;
+        u_avg[3]*u_avg[5]/(g*u_avg[1]*u_avg[4]*drho);;
+      -u_avg[5]/(g*u_avg[4]*drho);;
+        (g*rho1*u_avg[4]^3 - g*rho2*u_avg[4]^3 - rho2*u_avg[5]^2)/(g*rho2*u_avg[4]^2*drho);;
+      -u_avg[5]*u_avg[6]/(g*u_avg[4]^2*drho)];
+      [u_avg[6]/(g*u_avg[4]*drho);;
+        u_avg[2]*u_avg[6]/(g*u_avg[1]*u_avg[4]*drho);;
+        u_avg[3]*u_avg[6]/(g*u_avg[1]*u_avg[4]*drho);;
+      -u_avg[6]/(g*u_avg[4]*drho);;
+      -u_avg[5]*u_avg[6]/(g*u_avg[4]^2*drho);;
+        (g*rho1*u_avg[4]^3 - g*rho2*u_avg[4]^3 - rho2*u_avg[6]^2)/(g*rho2*u_avg[4]^2*drho)]]
```

Comment:
Similar to comment above. Should exploit symmetry to make this computation cheaper. Also, a comparison between this dissipation term to the specialized `(dissipation::DissipationLocalLaxFriedrichs)` would be informative.